### ISSUE TITLE
this.logDb.close is not a function

### ISSUE THREAD

**Issue by Bendernnm** [winstonjs/winston-mongodb#139]
> "winston": "3.0.0",
> "winston-mongodb": "4.0.3"

1) Config logger's db transport. 
2) Success start server, connect to db.
3) Also success write log to db, after start. 
4) Then dissconnect db and catch unhandledRejection with reason: 

> TypeError: this.logDb.close is not a function
    at exports.MongoDB.MongoDB.close (P:\project\node_modules\winston-mongodb\lib\winston-mongodb.js:167:14)
    at exports.MongoDB.TransportStream.once.src (P:\project\node_modules\winston-transport\index.js:49:14)
    at Object.onceWrapper (events.js:317:30)
    at emitTwo (events.js:131:20)
    at exports.MongoDB.emit (events.js:214:7)
    at DerivedLogger.Readable.unpipe (P:\project\node_modules\readable-stream\lib\_stream_readable.js:733:8)
    at unpipe (P:\project\node_modules\readable-stream\lib\_stream_readable.js:661:9)
    at exports.MongoDB.onerror (P:\project\node_modules\readable-stream\lib\_stream_readable.js:638:5)
    at emitOne (events.js:121:20)
    at exports.MongoDB.emit (events.js:211:7)
    at logDb.collection.insertOne.then.catch.err (P:\project\node_modules\winston-mongodb\lib\winston-mongodb.js:207:12)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

Crutch-solution (bad solution), in file where I configurate my logger: 

```
const WinstonMongoDB = require('winston-mongodb');

WinstonMongoDB.MongoDB.prototype.close =  function() {
    if (!this.logDb || typeof this.logDb.close !== 'function') {
        return;
    }

    this.logDb.close().then(()=>this.logDb = null).catch(err=>{
        console.error('Winston MongoDB transport encountered on error during '
            + 'closing.', err);
    });
};

```
Also tried change version to 3.0.2, but got other issues and deprications.

What do you think about it? 
Maybe you have better decision or you can explain why I get this error?

Will it fix in next patch?

**Comment by jjm340** [comment]
My issue is slightly different, even after I close the logger instance by using .close() the connection remains open and the app hangs - not sure how to manually close the connection

**Comment by yurijmikhalevich** [comment]
Closed as duplicate to #130
